Kbs.Wpf/Boat/Details/BoatDetailPage.xaml.cs

```diff
@@ -50,4 +72,50 @@ private void ReservationSelected(object sender, MouseButtonEventArgs e)
 
         _navigationManager.Navigate(() => new ViewPageSpecific(dataContext.ReservationId, _navigationManager));
     }
+
+    private void RequestDeletion(object sender, RoutedEventArgs e)
+    {
+        DateTime? newDeleteRequestDateValue;
+        string popupMessage = "";
+        if (ViewModel.DeleteRequestDate == null)
+        {
+            newDeleteRequestDateValue = DateTime.Now;
+            popupMessage = $"Verwijdering is aangevraagd.\nHet duurt {_daysForDeletion} dagen voordat je deze boot permanent mag verwijderen.";
+        }
+        else
+        {
+            newDeleteRequestDateValue = null;
+            popupMessage = "Verwijdering aanvraag is ongedaan gemaakt.";
+        }
+        ViewModel.DeleteRequestDate = newDeleteRequestDateValue;
+        MessageBox.Show(popupMessage);
+        ViewModel.Status = BoatStatus.Maintaining.ToDutchString();
+        BoatEntity newBoatValues = new BoatEntity() { BoatId = ViewModel.BoatId, BoatTypeId = ViewModel.BoatTypeId, Name = ViewModel.Name, DeleteRequestDate = ViewModel.DeleteRequestDate, Status = BoatStatus.Maintaining };
+        _boatRepository.Update(newBoatValues);
+        _navigationManager.Navigate(() => new BoatDetailPage(_navigationManager, ViewModel.BoatId));
```

Comment:
Als dit bedoelt is om te resetten, zet er even een comment bij, of reset de veranderde properties